backend/app/api/endpoints/rag.py

```
@@ -307,278 +74,43 @@ async def retrieve_documents(
+            result = await LocalRagGateway().query(runtime_spec, db=db)
 
-        return IndexNameResponse(
-            knowledge_id=knowledge_id,
-            index_name=index_name,
-            index_strategy=index_strategy,
-        )
+        return {"records": result.get("records", [])}
     except HTTPException:
         raise
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)